apisix/plugins/redirect.lua

```diff
@@ -35,10 +36,26 @@ local schema = {
     properties = {
         ret_code = {type = "integer", minimum = 200, default = 302},
         uri = {type = "string", minLength = 2, pattern = reg},
+        regex_uri = {
+            description = "new uri that substitute from client uri " ..
+                    "for upstream, lower priority than uri property",
+            type        = "array",
+            maxItems    = 2,
+            minItems    = 2,
+            items       = {
+                description = "regex uri",
+                type = "string",
+            }
+        },
         http_to_https = {type = "boolean"},
     },
     oneOf = {
-        {required = {"uri"}},
+        {
+            anyOf = {
```

Comment:
What about using `oneOf` so `uri` and `regex_uri` cannot co-exist.

Reply:
I think so